app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,12 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
```

Comment:
Oh sorry, I see this concern is only included into the ApplicationController + other base controllers, that'd make sense.

So why we do need it at all?  It looks like we're basically getting "somemodel_params" methods in every controller so we can call it in each controller to parse params.  Could we simply have a fixed method in the base which has varying behaviour?  No need to redefine it with a different name as far as I can see.

<pre>
def safe_params
  params.require(controller_name.singularize.to_sym).
    permit(*resource_class.permitted_attributes)
end
</pre>

And use `safe_params` instead of `puppetclass_params`.